ArrayOperationTests/SortTests.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+using ArrayOperations;
+
+namespace ArrayOperationTests
+{
+  public class SortTests
+  {
+    [Fact]
+    public void Sort_Should_Throw_ArgumentException_When_Input_Array_Is_Empty()
+    {
+      var input = new int[] {};
+      
+      Action actual = () => Arrays.Sort(input);
+      
+      Assert.Throws<ArgumentException>(actual);
+    }
+
+    [Fact]
+    public void Sort_Should_Throw_ArgumentNullException_When_Input_Is_Null()
+    {
+      int[] input = null;
+      
+      Action actual = () => Arrays.Sort(input);
+      
+      Assert.Throws<ArgumentNullException>(actual);
+    }
+
+    [Theory]
+    [InlineData(new [] {9001})]
+    [InlineData(new [] {0})]
+    public void Sort_Should_Return_Input_When_Array_Has_One_Element(int[] input)
+    {
+      int[] actual;
+      
+      actual = Arrays.Sort(input);
+      
+      Assert.Equal(input,actual);
+    }
+
+    [Theory]
+    [InlineData(new[] {1, 0}, new[] {0, 1})]
+    [InlineData(new[] {0, 1}, new[] {0, 1})]
+    [InlineData(new[] {1, 0, 2, -1, 3}, new[] {-1, 0, 1, 2, 3})]
+    [InlineData(new[] {1, -1, 1}, new[] {-1, 1, 1})]
+    [InlineData(new[] {1, 1, 1}, new[] {1, 1, 1})]
+    public void Sort_Should_Return_Expected_Array(int[] input, int[] expected)
```

Comment:
Not a review comment so much but something to make you think.

What you're doing here is very common.  You supply the input and the expected output.

Another way to do this kind of testing is to supply an input and a verification algorithm.  For example, for a sorting algorithm you can verify by taking the result and making sure each item is >= the previous.  Then you don't have to supply the expected output.

Anyways, what you have is probably best.  I'm just rambling.

Reply:
It's definitely not easy to read. I don't like how the inline data looks, even more so for the other tests that have more arguments. I saw there other ways to do it but it requires more OOP knowledge than I currently poses. One thing I've learned is that it's not a good idea to introduce logic in the actual tests.